integration-tests/startup.spec.js

```diff
@@ -9,153 +9,168 @@ const {
 
-describe('startup', () => {
-  let agent
-  let proc
-  let sandbox
-  let cwd
-  let startupTestFile
-
-  before(async () => {
-    sandbox = await createSandbox()
-    cwd = sandbox.folder
-    startupTestFile = path.join(cwd, 'startup/index.js')
-  })
-
-  after(async () => {
-    await sandbox.remove()
-  })
-
-  context('programmatic', () => {
-    beforeEach(async () => {
-      agent = await new FakeAgent().start()
+const execArgvs = [
+  [],
+  ['--import', 'dd-trace/register'],
+  ['--loader', 'dd-trace/loader-hook.mjs'],
+  ['--loader', 'dd-trace/loader-hook']
+]
+
+execArgvs.forEach((execArgv) => {
+  describe(`startup ${execArgv.join(' ')}`, () => {
+    let agent
+    let proc
+    let sandbox
+    let cwd
+    let startupTestFile
+
+    before(async () => {
+      sandbox = await createSandbox()
+      cwd = sandbox.folder
+      startupTestFile = path.join(cwd, 'startup/index.js')
     })
 
-    afterEach(async () => {
-      proc.kill()
-      await agent.stop()
+    after(async () => {
+      await sandbox.remove()
     })
 
-    it('works for options.port', async () => {
-      proc = await spawnProc(startupTestFile, {
-        cwd,
-        env: {
-          AGENT_PORT: agent.port
-        }
-      })
-      return curlAndAssertMessage(agent, proc, ({ headers, payload }) => {
-        assert.propertyVal(headers, 'host', `127.0.0.1:${agent.port}`)
-        assert.isArray(payload)
-        assert.strictEqual(payload.length, 1)
-        assert.isArray(payload[0])
-        assert.strictEqual(payload[0].length, 1)
-        assert.propertyVal(payload[0][0], 'name', 'web.request')
+    context('programmatic', () => {
+      beforeEach(async () => {
+        agent = await new FakeAgent().start()
       })
-    })
 
-    it('works for options.url', async () => {
-      proc = await spawnProc(startupTestFile, {
-        cwd,
-        env: {
-          AGENT_URL: `http://localhost:${agent.port}`
-        }
+      afterEach(async () => {
+        proc.kill()
+        await agent.stop()
       })
-      return curlAndAssertMessage(agent, proc, ({ headers, payload }) => {
-        assert.propertyVal(headers, 'host', `localhost:${agent.port}`)
-        assert.isArray(payload)
-        assert.strictEqual(payload.length, 1)
-        assert.isArray(payload[0])
-        assert.strictEqual(payload[0].length, 1)
-        assert.propertyVal(payload[0][0], 'name', 'web.request')
-      })
-    })
-  })
 
-  context('env var', () => {
-    beforeEach(async () => {
-      agent = await new FakeAgent().start()
-    })
-
-    afterEach(async () => {
-      proc.kill()
-      await agent.stop()
-    })
-
-    it('works for DD_TRACE_AGENT_PORT', async () => {
-      proc = await spawnProc(startupTestFile, {
-        cwd,
-        env: {
-          DD_TRACE_AGENT_PORT: agent.port
-        }
+      it('works for options.port', async () => {
+        proc = await spawnProc(startupTestFile, {
+          cwd,
+          execArgv,
```

Comment:
Alternatively, env.NODE_OPTIONS could be set below. I'm not sure which approach is preferred. execArgv seems more correct but both approaches seem to run the loader.